manager/integration/tests/test_snapshot_check.py

```diff
@@ -0,0 +1,425 @@
+import os
+import subprocess
+import time
+import datetime
+import json
+
+from common import client, volume_name  # NOQA
+from common import update_setting
+from common import get_self_host_id
+from common import create_and_check_volume, wait_for_volume_healthy
+from common import cleanup_volume
+from common import wait_for_rebuild_start, wait_for_rebuild_complete
+from common import create_snapshot
+from common import settings_reset # NOQA
+
+from test_basic import get_volume_endpoint
+
+from common import Gi, Mi
+from common import RETRY_COUNTS, RETRY_INTERVAL
+from common import SETTING_SNAPSHOT_DATA_INTEGRITY
+from common import SETTING_SNAPSHOT_DATA_INTEGRITY_IMMEDIATE_CHECK_AFTER_SNAPSHOT_CREATION  # NOQA
+from common import SETTING_SNAPSHOT_DATA_INTEGRITY_CRONJOB
+from common import SNAPSHOT_DATA_INTEGRITY_IGNORED
+from common import SNAPSHOT_DATA_INTEGRITY_ENABLED
+from common import SNAPSHOT_DATA_INTEGRITY_FAST_CHECK
+from common import SNAPSHOT_DATA_INTEGRITY_DISABLED
+
+
+def test_snapshot_hash_global_enabled_with_immediate_hash(client, volume_name, settings_reset):  # NOQA
+    update_settings(client, SNAPSHOT_DATA_INTEGRITY_ENABLED, "true")
+    check_hash_calculated_and_with_immediate_hash(client, volume_name,
+                                                  SNAPSHOT_DATA_INTEGRITY_IGNORED)  # NOQA
+
+
+def test_snapshot_hash_global_enabled_without_immediate_hash(client, volume_name, settings_reset):  # NOQA
+    update_settings(client, SNAPSHOT_DATA_INTEGRITY_ENABLED, "false")
+    check_hash_calculated_and_without_immediate_hash(client, volume_name,
+                                                     SNAPSHOT_DATA_INTEGRITY_IGNORED)  # NOQA
+
+
+def test_snapshot_hash_global_fast_check_with_immediate_hash(client, volume_name, settings_reset):  # NOQA
+    update_settings(client, SNAPSHOT_DATA_INTEGRITY_FAST_CHECK, "true")
+    check_hash_calculated_and_with_immediate_hash(client, volume_name, SNAPSHOT_DATA_INTEGRITY_IGNORED)   # NOQA
+
+
+def test_snapshot_hash_global_fast_check_without_immediate_hash(client, volume_name, settings_reset):  # NOQA
+    update_settings(client, SNAPSHOT_DATA_INTEGRITY_FAST_CHECK, "false")
+    check_hash_calculated_and_without_immediate_hash(client, volume_name,
+                                                     SNAPSHOT_DATA_INTEGRITY_IGNORED)  # NOQA
+
+
+def test_snapshot_hash_global_disabled_with_immediate_hash(client, volume_name, settings_reset):  # NOQA
```

Comment:
Does it mean that the immediate_hash depends on the `snapshot_data_integrity`? If `snapshot_data_integrity` is disabled, immediate_hash will not take effect?

Reply:
Yes. The `snapshot_data_integrity` is like a main option. The `immediate_hash` and `cronjob` are like suboptions.